gson/src/main/java/com/google/gson/reflect/ReflectionAccessUtils.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2017 The Gson authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gson.reflect;
+
+import com.google.gson.reflect.impl.PreJava9ReflectionAccessor;
+import com.google.gson.reflect.impl.UnsafeReflectionAccessor;
+import com.google.gson.util.VersionUtils;
+
+import java.lang.reflect.AccessibleObject;
+
+/**
+ * Provides {@link ReflectionAccessor} instance which may be used to avoid reflective access issues
+ * appeared in Java 9, like {@link java.lang.reflect.InaccessibleObjectException} thrown or
+ * warnings like
+ * <pre>
+ *   WARNING: An illegal reflective access operation has occurred
+ *   WARNING: Illegal reflective access by ...
+ * </pre>
+ * <p/>
+ * Works both for Java 9 and earlier Java versions.
+ */
+public class ReflectionAccessUtils {
```

Comment:
Can you delete this class, and move this code to a getInstance() method under ReflectionAccessor?